### PR TITLE
Add Error LED to the Kasli's MasterBase and SatelliteBase submodule

### LINKED ISSUE #1947: Kasli Error LED does not light up on panic
# Bug Report

## One-Line Summary

The Kasli Error LED does not light up on an error, aka when the rust firmware panics. 
**EDIT: This is only the case for `Master` variants, for `Standalone` variants the Error LED lights up as usual.** 

### Steps to Reproduce

E.g. don't feed a clock to a  Kasli which expects an external reference at startup.

### Expected Behavior

The LED should light up.

### Actual (undesired) Behavior

```
Starting firmware.
[     0.000014s]  INFO(runtime): ARTIQ runtime starting...
[     0.003938s]  INFO(runtime): software ident 7.8116.eba143a;npl9master
[     0.010390s]  INFO(runtime): gateware ident 7.8116.eba143a;npl9master
[     0.016845s]  INFO(runtime): log level set to INFO by default
[     0.022574s]  INFO(runtime): UART log level set to INFO by default
[     0.139378s]  WARN(runtime::rtio_clocking): rtio_clock setting not recognised. Falling back to default.
[     0.147478s]  WARN(runtime::rtio_clocking): si5324_ext_ref and ext_ref_frequency compile-time options are deprecated. Please use the rtio_clock coreconfig settings instead.
[     0.162890s]  INFO(runtime::rtio_clocking): using 10MHz reference to make 125MHz RTIO clock with PLL
panic at runtime/rtio_clocking.rs:225:55: cannot initialize Si5324: "Si5324 misses clock input signal"
backtrace for software version 7.8116.eba143a;npl9master:
0x4001e660
0x4000aca0
0x4000a2a0
0x4001cf70
0x4000770c
0x4002fcc0
0x4002fc68
0x4001d7e8
halting.
use `artiq_coremgmt config write -s panic_reset 1` to restart instead
```

Error LED doesn't turn on.

### Your System (omit irrelevant parts)

* ARTIQ version: ARTIQ-7 [eba143a](https://github.com/m-labs/artiq/commit/eba143a47530adbd9d402a9736d0fdc06dcf836f)

### PR DESCRIPTION
**Pull request by cc78078**
# ARTIQ Pull Request

## Description of Changes
add Error LED to the Kasli's MasterBase and SatelliteBase submodule.

Test setup
1. Get a Kasli board
2. Insert a `panic!()` statement into the main function of files artiq/firmware/runtime/main.rs and artiq/firmware/satman/main.rs to simulate the rust firmware panic.

| before changes | after changes |
|---------------- | -----------------|
| Error LED Off | Error LED On |

### Related Issue
Closes #1947 

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

### All Pull Requests

- [x] Use correct spelling and grammar.
- [x] Close/update issues.

### Code Changes

- [x] Test your changes or have someone test them. Mention what was tested and how.

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

**Comment by sbourdeauducq** [comment]
We may want this in satellites as well.

**Comment by jordens** [comment]
In the case of satellites it would seem useful/intuitive to use the error_led to indicate missing DRTIO uplink.

**Comment by sbourdeauducq** [comment]
I'm not sure about this:
* There is already an LED near the SFP to indicate link status.
* With distributed processing there might be distributed idle/startup kernels as well someday, which potentially would be run in case of missing uplink.
* It is useful (and consistent) to use the ERROR LEDs for more serious conditions.

File: artiq/gateware/targets/kasli.py [hunk 1]
```diff
@@ -462,6 +465,9 @@ def __init__(self, rtio_clk_freq=125e6, enable_sata=False, *, with_wrpll=False,
         disable_cdr_clk_ibuf = Signal(reset=1)
         disable_cdr_clk_ibuf.attr.add("no_retiming")
         if self.platform.hw_rev == "v2.0":
+            self.submodules.error_led = gpio.GPIOOut(Cat(
+                self.platform.request("error_led")))
+            self.csr_devices.append("error_led")
```
Comment (sbourdeauducq) [inline artiq/gateware/targets/kasli.py:470]: Merged a bit too fast.

This LED code does not belong with the transceiver clocking code.

**Comment by jordens** [comment]
Ack

**Pull request merged**